doc/source/whatsnew/v0.23.2.txt

```diff
@@ -65,7 +65,7 @@ Bug Fixes
 **I/O**
 
 - Bug in :func:`read_csv` that caused it to incorrectly raise an error when ``nrows=0``, ``low_memory=True``, and ``index_col`` was not ``None`` (:issue:`21141`)
--
+- Bug in :func:`json_normalize` where passing an array of values and a `record_prefix` would raise a `TypeError` (:issue:`21536`)
```

Comment:
use double backticks on ``record_prefix`` and ``TypeError``

Reply:
isn't the issue that this is for non-string columns?

Reply:
Will change to double backticks.
The issue is not because the column is non-string; it is because the array contains values (i.e., not objects or arrays). Suggestions are welcome.

Reply:
how so? before concatting integers and strings would fail.

Reply:
I misunderstood. When you mentioned *columns*, I thought they were the data columns, not the **column names**. Yes. The bug is caused by concatenating non-string column names, which happens when we pass an array of values (integers, strings, etc.). I can't think of other paths to trigger the bug.

My note explains how one can reproduce the bug. Looks like you want it to describe the reason.
I'm fine with either. How about this?
- Bug in :func:`json_normalize` that caused it to incorrectly raise an error when concatenating non-string columns (:issue:`21536`)

Reply:
closer, though a user won't know what concatenating is (in this context), maybe just say something like

bug in formatting the ``record_prefix`` with integer columns